subprojects/gradle/kubernetes/src/main/kotlin/com/avito/utils/gradle/Kubernetes.kt

```diff
@@ -40,11 +40,27 @@ fun createKubernetesClient(
                     caCertFile = kubernetesCredentials.caCertFile
                 }
 
-                // working with multiple namespaces/contexts is not supported
-                require(getNamespace() == namespace) {
-                    "kubernetes.context.namespace should be $namespace, but was ${getNamespace()}. " +
+                val namespaceFromKubeConfig = getNamespace()
```

Comment:
`...FromKubeConfig` - we add the context because we are inside `config.apply {}`
Maybe, without `apply` the code will be a bit more obvious and less nested.

```kotlin
val config = Config.fromKubeconfig()
val namespace = config.getNamespace()
....
```